### PR TITLE
Fixed: restartable MessageHistoryService

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
`Service`s can be restarted via `Client.pause/resume`. It was not possible with `MessageHistoryService`, which was derived from `threading.Thread`.

### CODE REVIEW

**Comment by etam** [comment]
Maybe it would be cleaner to use composition, instead of inheriting from Thread? This delayed initialization of Thread class looks like a dirty hack.

**Comment by mfranciszkiewicz** [comment]
@etam Either way I'd have to re-create the thread. Unless "stop" doesn't stop the service.

**Comment by etam** [comment]
@mfranciszkiewicz recreating Thread object is much cleaner solution.

**Review by etam — APPROVED** [review]

File: golem/network/history.py [hunk 1]
```diff
@@ -43,33 +43,41 @@ class MessageHistoryService(IService, threading.Thread):
...
-        return self._started.is_set() and not self._stop_event.is_set()
+        return (
+            self._thread and
+            self._thread.is_alive() and
+            not self._stop_event.is_set()
+        )
 
     def start(self) -> None:
-        threading.Thread.start(self)
+        if not self.running:
```
Comment (jiivan) [inline golem/network/history.py]:
```python
if self.running:
    return
```
?

**Review by jiivan — APPROVED** [review]
LGTM. One suggestion.

**Pull request merged**